File: vm_Assignments/vmc/L2.2/RocketShipFactored.c
#include<stdio.h>
void cone_part()
{
    printf("    ^\n");
    printf("   /|\\\n");
    printf("  //|\\\\ \n");
    printf(" ///|\\\\\\ \n");
}
void body()
{
    for(int i=0; i<4; i++)
    printf("+*******+\n");
    printf("+-------+\n");
    for(int i=0; i<4; i++)
    printf("+*******+\n");
    printf("+-------+\n");
}
void make_rocket()
{
    cone_part();
    body();
    cone_part();
}
void main()
{
    make_rocket();
}

File: vm_Assignments/vm/Beginning/L2.2/RocketShipFactored.py
def cone_part():
    print(r"    ^    ")
    print(r"   /|\   ")
    print(r"  //|\\  ")
    print(r" ///|\\\ ")

def minus_sign():
    print("+-------+")

def star_sign():
    print("+*******+")

def body():
    minus_sign()
    for i in range(4):
        star_sign()
    minus_sign()
    for i in range(4):
        star_sign()
    minus_sign()

def make_rocket():
    cone_part()
    body()
    cone_part()

make_rocket()
